pkg/publish/default.go

```diff
@@ -260,3 +260,25 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 func (d *defalt) Close() error {
 	return nil
 }
+
+// Prepare implements publish.Preparer
+func (d *defalt) Prepare(ctx context.Context, s string) error {
+	var no []name.Option
+	if d.insecure {
+		no = append(no, name.Insecure)
+	}
+
+	for _, tagName := range d.tags {
+		tag, err := name.NewTag(fmt.Sprintf("%s:%s", d.namer(d.base, s), tagName), no...)
+		if err != nil {
+			return err
+		}
+
+		if err := remote.CheckPushPermission(tag, d.keychain, d.t); err != nil {
+			return err
+		}
+	}
```

Comment:
Auth is generally configured at the repo level, and all these tags will have the same repo, so we can just check the first one, or `:latest`.

It looks like `CheckPushPermission` should have just taken a `name.Repository` anyway, it doesn't use the tag: https://github.com/google/go-containerregistry/blob/v0.13.0/pkg/v1/remote/check.go#L34

So let's just check the first tag and not bother checking each one.

Reply:
that makes sense.